### PR TITLE
TG-979 - feature to deploy the ingestion spec into 2 different cluster

### PR DESCRIPTION
**Pull request by Kaali09**
This PR includes the changes to deploy the ingestion spec into 2 different cluster by just selecting the ingestion spec name from jenkins job.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ansible/druid-ingestion.yml [hunk 1]
```diff
@@ -1,6 +1,56 @@
 ---
+- hosts: rollup-overlord
+  become: yes
+  pre_tasks:
```
Comment (keshavprasadms) [inline ansible/druid-ingestion.yml:4]: Move this entire block as a separate play running against `localhost`, so we can remove duplication of code.
```
- hosts: localhost
  tasks:
   - name: set_fact
     set_fact:
       single_cluster: true
     tags: populate_var

   - name: check for hosts
     set_fact:
       single_cluster: false
     when: "'rollup-overlord' in groups"
     tags: populate_var
```

File: ansible/druid-ingestion.yml [hunk 2]
```diff
@@ -1,6 +1,56 @@
...
+    - role: druid-ingestion
+      vars:
+        ingestion_task_names: "{{ rollup_ingestion_task_names }}"
+        overlord_host: "{{ rollup_overlord_host }}"
+      when: not single_cluster
 
-- hosts: "{{ cluster }}-overlord"
+- hosts: raw-overlord
   become: yes
+  pre_tasks:
```
Comment (keshavprasadms) [inline ansible/druid-ingestion.yml:29]: This wont be required after implementing previous comment

File: pipelines/ops/druid/Jenkinsfile.ingestion [hunk 3]
```diff
@@ -25,8 +25,22 @@ node("ops-slave") {
                 module = sh(returnStdout: true, script: "echo $JOB_NAME").split('/')[-2].trim()
                 jobName = sh(returnStdout: true, script: "echo $JOB_NAME").split('/')[-1].trim()
                 currentWs = sh(returnStdout: true, script: 'pwd').trim()
+        // populating the vars for raw-overlord and rollup-overlord cluster respectively.
+                list_vars="$ingestion_task_names";
```
Comment (keshavprasadms) [inline pipelines/ops/druid/Jenkinsfile.ingestion:29]: Change to `list_vars=params.ingestion_task_names;`

File: pipelines/ops/druid/Jenkinsfile.ingestion [hunk 4]
```diff
@@ -25,8 +25,22 @@ node("ops-slave") {
...
+                println(list_vars);
+                List<String> list_vars_array = list_vars.split(',');
```
Comment (keshavprasadms) [inline pipelines/ops/druid/Jenkinsfile.ingestion:31]: List<String> task_names_list = task_names.split(',');

File: pipelines/ops/druid/Jenkinsfile.ingestion [hunk 5]
```diff
@@ -25,8 +25,22 @@ node("ops-slave") {
...
+                List<String> rollup_vars = [], raw_vars = [];
+                list_vars_array.each {
+                    if(it.startsWith("rollup")) {
+                        rollup_vars.add(it);
+                    }
+                    else
+                       raw_vars.add(it);
+                }
+                rollup_ingestion_task_names = rollup_vars.join(",");
```
Comment (keshavprasadms) [inline pipelines/ops/druid/Jenkinsfile.ingestion:40]: We dont need additonal variables `rollup_ingestion_task_names` and `raw_ingestion_task_names`, you can remove this and the next line.

File: pipelines/ops/druid/Jenkinsfile.ingestion [hunk 6]
```diff
@@ -25,8 +25,22 @@ node("ops-slave") {
...
+                raw_ingestion_task_names = raw_vars.join(",");
                 ansiblePlaybook = "${currentWs}/ansible/druid-ingestion.yml"
-                ansibleExtraArgs = "--extra-vars \"ingestion_task_names=${params.ingestion_task_names} cluster=${params.cluster}\" --tags ${params.action} --vault-password-file /var/lib/jenkins/secrets/vault-pass -vv"
+                ansibleExtraArgs = "--extra-vars \"rollup_ingestion_task_names=$rollup_ingestion_task_names raw_ingestion_task_names=$raw_ingestion_task_names\" --tags populate_var,${params.action} --vault-password-file /var/lib/jenkins/secrets/vault-pass -vv"
```
Comment (keshavprasadms) [inline pipelines/ops/druid/Jenkinsfile.ingestion:43]: Directly pass the original variables `rollup_vars.join(",")` and `raw_vars.join(",")` in `--extra-vars`

File: ansible/druid-ingestion.yml (hunk 1 above)
Comment (Kaali09) [reply ansible/druid-ingestion.yml:4]: This variable single_cluster is not available for next play. As we are setting this variable for loacalhost host group, it is not available for next play when we run against rollup-overload or raw-overlord. That is why I added this set_fact in each play.
@keshavprasadms

File: ansible/druid-ingestion.yml (hunk 2 above)
Comment (Kaali09) [reply ansible/druid-ingestion.yml:29]: Ansible is not taking the variable from previous play.

File: pipelines/ops/druid/Jenkinsfile.ingestion (hunk 3 above)
Comment (Kaali09) [reply pipelines/ops/druid/Jenkinsfile.ingestion]: Thanks. updated as suggested.

File: pipelines/ops/druid/Jenkinsfile.ingestion (hunk 4 above)
Comment (Kaali09) [reply pipelines/ops/druid/Jenkinsfile.ingestion]: updated as suggested.

File: pipelines/ops/druid/Jenkinsfile.ingestion (hunk 5 above)
Comment (Kaali09) [reply pipelines/ops/druid/Jenkinsfile.ingestion]: updated as suggested.

File: pipelines/ops/druid/Jenkinsfile.ingestion (hunk 6 above)
Comment (Kaali09) [reply pipelines/ops/druid/Jenkinsfile.ingestion]: yeah it worked. The curly braces did a trick. Earlier I tried with out curly braces did not work. 

Updated as suggested.

File: ansible/druid-ingestion.yml (hunk 1 above)
Comment (keshavprasadms) [reply ansible/druid-ingestion.yml:4]: @Kaali09 Can you try to use this in the when condition?
`{{ hostvars['localhost']['single_cluster'] }}`

Comment (Kaali09) [reply ansible/druid-ingestion.yml]: thanks . it worked without curly braces.

**Review by keshavprasadms — APPROVED** [review]

**Review by anandp504 — APPROVED** [review]
Looks good :+1:

**Pull request merged**